components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/observer/AnalyticsMediationFlowObserver.java

```diff
@@ -38,13 +39,17 @@ public class AnalyticsMediationFlowObserver implements MessageFlowObserver, Tena
     public AnalyticsMediationFlowObserver(List<String> publisherTypes) {
         if (publisherTypes != null && !publisherTypes.isEmpty() &&
                 (publisherTypes.contains(MediationDataPublisherConstants.DATABRIDGE_PUBLISHER_TYPE )
-                        || publisherTypes.contains(MediationDataPublisherConstants.LOG_PUBLISHER_TYPE))) {
+                        || publisherTypes.contains(MediationDataPublisherConstants.LOG_PUBLISHER_TYPE)
+                        || publisherTypes.contains(MediationDataPublisherConstants.CHOREO_PUBLISHER_TYPE))) {
             if (publisherTypes.contains(MediationDataPublisherConstants.DATABRIDGE_PUBLISHER_TYPE)) {
                 statPublishers.add(EIStatisticsPublisher.GetInstance());
             }
             if (publisherTypes.contains(MediationDataPublisherConstants.LOG_PUBLISHER_TYPE)) {
                 statPublishers.add(ElasticStatisticsPublisher.GetInstance());
             }
+            if (publisherTypes.contains(MediationDataPublisherConstants.CHOREO_PUBLISHER_TYPE)) {
+                statPublishers.add(ChoreoStatisticsPublisher.GetInstance());
+            }
         } else {
```

Comment:
Shall we optimize this if condition, something like:
```java
        boolean useDefaultPublisher = true;
        if (publisherTypes != null && !publisherTypes.isEmpty()) {
            if (publisherTypes.contains(MediationDataPublisherConstants.DATABRIDGE_PUBLISHER_TYPE)) {
                statPublishers.add(EIStatisticsPublisher.GetInstance());
                useDefaultPublisher = false;
            }
            // Other publisherTypes go here
        }

        if (useDefaultPublisher) {
            statPublishers.add(ElasticStatisticsPublisher.GetInstance());
        }
```

Reply:
Optimised as requested by https://github.com/wso2/micro-integrator/pull/2927/commits/7ce08a2dbe17617040a9584d9afe40c25a4178fc